scipy/stats/_distn_infrastructure.py

```diff
@@ -3836,3 +3884,423 @@ def get_distribution_names(namespace_pairs, rv_base_class):
         if isinstance(value, rv_base_class):
             distn_names.append(name)
     return distn_names, distn_gen_names
+
+
+def _combine_bounds(name, bounds_type, user_bounds, shape_domain,
+                    integral):
+    """Intersection of user-defined bounds and distribution PDF/PMF domain"""
+
+    user_bounds = np.atleast_1d(user_bounds)
+
+    if user_bounds.shape != (2,):
+        # `shape_bounds` is already checked, so this can only happen with
+        # `loc_bounds` and `scale_bounds`
+        message = (f"`{bounds_type}` must be a tuple containing exactly "
+                   "two elements: the lower and upper bound of the "
+                   f"`{name}` parameter.")
+        raise ValueError(message)
+
+    if user_bounds[0] > user_bounds[1]:
+        message = (f"There are no values for `{name}` on the interval "
+                   f"{list(user_bounds)}.")
+        raise ValueError(message)
+
+    bounds = (max(user_bounds[0], shape_domain[0]),
+              min(user_bounds[1], shape_domain[1]))
+
+    if integral and (np.ceil(bounds[0]) > np.floor(bounds[1])):
+        message = (f"There are no integer values for `{name}` on the interval "
+                   f"defined by the user-provided bounds and the domain "
+                   "of the distribution.")
+        raise ValueError(message)
+    elif not integral and (bounds[0] > bounds[1]):
+        message = (f"There are no values for `{name}` on the interval "
+                   f"defined by the user-provided bounds and the domain "
+                   "of the distribution.")
+        raise ValueError(message)
+
+    if not np.all(np.isfinite(bounds)):
+        message = (f"The intersection of user-provided bounds for `{name}` "
+                   f"and the domain of the distribution is not finite. Please "
+                   f"provide finite bounds for shape `{name}` in "
+                   f"`{bounds_type}`.")
+        raise ValueError(message)
+
+    return bounds
+
+
+class FitResult:
+    r"""Result of fitting a discrete or continuous distribution to data
+
+    Attributes
+    ----------
+    dist : scipy.stats.rv_continuous or scipy.stats.rv_discrete
+        The distribution of which parameters were fit.
+    data : 1D array_like
+        The data to which the distribution was fit.
+    nllf : callable
+        The negative log-likehood function for the given `data`. Accepts a
+        single tuple containing the shapes, location, and scale of the
+        distribution.
```

Comment:
We can consider removing these, but it may be handy to encapsulate all the information that led to the fit result.